tests/unit/cluster/failure-marking.tcl

```diff
@@ -124,3 +114,67 @@ tags {external:skip tls:skip cluster singledb} {
         }
     }
 }
+
+# Test that no new failure-report is added once the node is already marked as FAIL
+start_cluster 3 1 {tags {external:skip cluster}} {
+    test "Primaries do not add failure-report after replica is already marked FAIL" {
+        # Primary nodes
+        set primary0 [srv 0 client];
+        set primary0_pid [srv 0 pid]
+        set primary1 [srv -1 client];
+        set primary1_pid [srv -1 pid]
+        set primary2 [srv -2 client];
+        set primary2_pid [srv -2 pid]
+
+        # Replica node
+        set replica0 [srv -3 client];
+        set replica0_pid [srv -3 pid]
+        set replica0_id [dict get [cluster_get_myself 3] id]
+
+        assert {[lindex [$primary0 role] 0] eq {master}}
+        assert {[lindex [$primary1 role] 0] eq {master}}
+        assert {[lindex [$primary2 role] 0] eq {master}}
+        assert {[lindex [$replica0 role] 0] eq {slave}}
```

Comment:
You should basically never use assert, since it won't show the other state on failure.
```suggestion
        assert_equal [lindex [$primary0 role] 0] {master}
        assert_equal [lindex [$primary1 role] 0] {master}
        assert_equal [lindex [$primary2 role] 0] {master}
        assert_equal [lindex [$replica0 role] 0] {slave}
```